Compiler warnings in "es-lang-c-stdc99.c"

After pulling the latest sources, clang generates the following warnings while compiling `es-lang-c-stdc99.c`:

```
dsl/es-lang-c-stdc99.c:1333:6: warning: assigning to 'char *' from 'unsigned char *' converts between pointers to integer types with different sign [-Wpointer-sign]
  bp = mio_memory_get_data (out, &size);
     ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
dsl/es-lang-c-stdc99.c:1406:6: warning: assigning to 'char *' from 'unsigned char *' converts between pointers to integer types with different sign [-Wpointer-sign]
  bp = mio_memory_get_data (out, &size);
     ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
dsl/es-lang-c-stdc99.c:2611:15: warning: expression result unused [-Wunused-value]
          /* free */(*s);
                     ^~
dsl/es-lang-c-stdc99.c:2621:26: warning: expression result unused [-Wunused-value]
          /* es_object_unref */(*o);
```

--- a/dsl/es-lang-c-stdc99.c
+++ b/dsl/es-lang-c-stdc99.c
@@ -1250,7 +1250,7 @@ djb2(unsigned char *str)
 	unsigned long hash = 5381;
 	int c;
 
-	while (c = *str++)
+	while ((c = *str++))
 		hash = ((hash << 5) + hash) + c; /* hash * 33 + c */
 
 	return hash;
@@ -2566,7 +2566,7 @@ recover(EsObject* fmt_object, va_list *aq)
 	  char **s;
 
 	  s = va_arg(*aq, char **);
-	  /* free */(*s);
+	  (void)/* free */(*s);
 
 	  *s = NULL;
 	}
@@ -2576,7 +2576,7 @@ recover(EsObject* fmt_object, va_list *aq)
 	  EsObject** o;
 
 	  o = va_arg(*aq, EsObject**);
-	  /* es_object_unref */(*o);
+	  (void)/* es_object_unref */(*o);
 	  *o = NULL;
 	}
     }

